### PR TITLE
Fix/11222 remove badge count from redeemed tests

### PR DESCRIPTION
**Pull request by puneetmahali1**
## Description
After scan the redeemed tests it was showing the badge count status.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11222

## Screenshots
https://user-images.githubusercontent.com/72390277/211859454-9ec53047-4d05-4169-8401-c5d0d9e40816.MP4

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 1]
```diff
@@ -27,6 +27,7 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 		vaccinationValueSetsProvider: VaccinationValueSetsProviding,
 		healthCertificateValidationOnboardedCountriesProvider: HealthCertificateValidationOnboardedCountriesProviding,
 		qrScannerCoordinator: QRScannerCoordinator,
+		badgeWrapper: HomeBadgeWrapper,
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:30]:
```suggestion
		homeBadgeWrapper: HomeBadgeWrapper,
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 2]
```diff
@@ -206,6 +208,7 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 	private let vaccinationValueSetsProvider: VaccinationValueSetsProviding
 	private let healthCertificateValidationOnboardedCountriesProvider: HealthCertificateValidationOnboardedCountriesProviding
 	private let qrScannerCoordinator: QRScannerCoordinator
+	private let badgeWrapper: HomeBadgeWrapper
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:211]:
```suggestion
	private let homeBadgeWrapper: HomeBadgeWrapper
```

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift [hunk 3]
```diff
@@ -41,7 +41,8 @@ class QRScannerCoordinator {
 		exposureSubmissionService: ExposureSubmissionService,
 		coronaTestService: CoronaTestServiceProviding,
 		familyMemberCoronaTestService: FamilyMemberCoronaTestServiceProviding,
-		recycleBin: RecycleBin
+		recycleBin: RecycleBin,
+		badgeWrapper: HomeBadgeWrapper
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift:45]:
```suggestion
		homeBadgeWrapper: HomeBadgeWrapper
```

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift [hunk 4]
```diff
@@ -98,6 +100,7 @@ class QRScannerCoordinator {
 	private let coronaTestService: CoronaTestServiceProviding
 	private let familyMemberCoronaTestService: FamilyMemberCoronaTestServiceProviding
 	private let recycleBin: RecycleBin
+	private let badgeWrapper: HomeBadgeWrapper
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift:103]:
```suggestion
	private let homeBadgeWrapper: HomeBadgeWrapper
```

**Review by flxschmidt — COMMENTED** [review]
LGTM. 👍

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 1 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:30]: Actually we are using the **badgeWrapper** in other part of the app that's why keep the same naming.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 2 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:211]: same here

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift (hunk 3 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift:45]: same

File: src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift (hunk 4 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift:103]: same as well

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 1 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:30]: Okay, your choice. It's just a tasty thing ;)

**Review by flxschmidt — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 5]
```diff
@@ -1794,6 +1797,8 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 				message: error.localizedDescription,
 				completion: { [weak self] in
 					self?.dismiss()
+					// reset the home badge count
+					self?.badgeWrapper.resetAll()
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: You are resetting the total count, that means, if there was a count before that was unrelated to this condition here, will also be reseted.

**Review by naveeddotio — CHANGES_REQUESTED** [review]
Please check my comment

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 5 above)
Comment (puneetmahali1) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: Fixed.

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**